src/stateAggregator/accessors/orgAccessor.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import { JsonMap, Nullable, Optional } from '@salesforce/ts-types';
+import { isEmpty } from '@salesforce/kit';
+import { AuthInfoConfig } from '../../config/authInfoConfig';
+import { Global } from '../../global';
+import { GlobalInfo } from '../globalInfoConfig';
+import { SfOrgs, SfOrg, SfInfoKeys } from '../types';
+import { AuthFields } from '../../org';
+import { ConfigContents, ConfigFile } from '../../exported';
+
+/**
+ * @deprecated
+ */
+export class GlobalInfoOrgAccessor {
+  public constructor(private globalInfo: GlobalInfo) {}
+
+  public getAll(decrypt = false): SfOrgs {
+    return this.globalInfo.get(SfInfoKeys.ORGS, decrypt);
+  }
+
+  public get(username: string, decrypt = false): Optional<SfOrg> {
+    const auth = this.globalInfo.get<SfOrg>(`${SfInfoKeys.ORGS}["${username}"]`, decrypt);
+    // For legacy, some things wants the username in the returned auth info.
+    if (auth && !auth.username) auth.username = username;
+    return auth;
+  }
+
+  public has(username: string): boolean {
+    return !!this.getAll()[username];
+  }
+
+  public set(username: string, org: SfOrg): void {
+    // For legacy, and to keep things standard, some things wants the username in auth info.
+    org.username ??= username;
+    this.globalInfo.set(`${SfInfoKeys.ORGS}["${username}"]`, org);
+  }
+
+  public update(username: string, org: Partial<SfOrg>): void {
+    // For legacy, and to keep things standard, some things wants the username in auth info.
+    org.username ??= username;
+    this.globalInfo.update(`${SfInfoKeys.ORGS}["${username}"]`, org);
+  }
+
+  public unset(username: string): void {
+    delete this.globalInfo.get(SfInfoKeys.ORGS)[username];
+  }
+}
+
+export abstract class BaseOrgAccessor<T extends ConfigFile, P extends ConfigContents> {
+  private configs: Map<string, Nullable<T>> = new Map();
```

Comment:
when might it be null/undefined?

Reply:
@mshanemc there are scenarios is AuthInfo were we begin constructing the auth file for a brand new auth before persisting it. In this case, there will be no config for it until the end of the process